semtypes/src/main/java/io/ballerina/types/definition/ListDefinition.java

```diff
@@ -113,19 +115,26 @@ private FixedLengthArray fixedLengthNormalize(FixedLengthArray array) {
             }
             i -= 1;
         }
-        return FixedLengthArray.from(initial.subList(0, i + 2), array.fixedLength);
+        return FixedLengthArray.from(initial.subList(0, i + 2), array.fixedLength());
     }
 
     private ComplexSemType createSemType(Env env, Atom atom) {
         BddNode bdd = BddCommonOps.bddAtom(atom);
-        ComplexSemType s = PredefinedType.basicSubtype(BasicTypeCode.BT_LIST, bdd);
-        this.semType = s;
-        return s;
+        ComplexSemType complexSemType = basicSubtype(BasicTypeCode.BT_LIST, bdd);
+        this.semType = complexSemType;
+        return complexSemType;
     }
 
-    public static SemType tuple(Env env, SemType... members) {
-        ListDefinition def = new ListDefinition();
-        return def.define(env, List.of(members));
+    public SemType define(Env env, List<CellSemType> initial) {
```

Comment:
shall we have `tupleTypeWrapped(Env env, SemType... members)` and `tupleTypeWrappedRo(Env env, SemType... members)`?